### PR TITLE
Refactoring conditional directives.

### PR DESCRIPTION
**Pull request by wmarquesr**
All fixed, i change the pull request for this branch.
I'll waiting for analisys.

### CODE REVIEW

File: kmod/igb/igb_main.c [hunk 1]
```diff
@@ -1483,13 +1484,16 @@ static void igb_update_mng_vlan(struct igb_adapter *adapter)
...
 #ifdef HAVE_VLAN_RX_REGISTER
-	    !vlan_group_get_device(adapter->vlgrp, old_vid)) {
+	old_vid_status = ((old_vid != (u16)IGB_MNG_VLAN_NONE)
+	                  && (vid != old_vid)
+										&& !vlan_group_get_device(adapter->vlgrp, old_vid));
 #else
-	    !test_bit(old_vid, adapter->active_vlans)) {
+  old_vid_status = ((old_vid != (u16)IGB_MNG_VLAN_NONE)
+	                  && (vid != old_vid)
+										&& !test_bit(old_vid, adapter->active_vlans));
```
Comment (pinealservo) [inline kmod/igb/igb_main.c]: I think we have a tab-vs-space issue going on here. The `&& (vid != old_vid)` line is lined up via spaces, but the following line has a whole lot of tabs that I assume were set at a much smaller value than the kernel standard of 8 spaces per tab. Per kernel standards, indentation should be done solely with tabs and a tab-width of 8. 

The original code was a bit off as well, as it used a combination of tabs and spaces, but if we're changing the lines anyway it's a perfect opportunity to fix it.

File: kmod/igb/igb_main.c [hunk 2]
```diff
@@ -7876,16 +7883,19 @@ static void igb_lro_receive(struct igb_q_vector *q_vector,
...
-		    igb_lro_hdr(lro_skb)->th.window != lroh->th.window) {
+     check_data_len = (check_data_len
+			                 || data_len != new_skb->data_len
+											 || skb_shinfo(new_skb)->nr_frags
+											 >= (MAX_SKB_FRAGS - skb_shinfo(lro_skb)->nr_frags));
+#endif
+     check_data_len = (check_data_len
+			                 || igb_lro_hdr(lro_skb)->th.ack_seq != lroh->th.ack_seq
+											 || igb_lro_hdr(lro_skb)->th.window != lroh->th.window);
+		if (check_data_len){
```
Comment (pinealservo) [inline kmod/igb/igb_main.c]: More extra tabs in this hunk.

**Comment by andrew-elder** [comment]
We need to get closure on the kernel igb vs open-avb discussion before we spend to much time correcting this commit here. Let me start an offline discussion with Intel guys.

**Comment by andrew-elder** [comment]
We are going to accept igb driver updates here on open-avb.

**Comment by andrew-elder** [comment]
Does this pull request replace #330? We should close #330 then?

**Comment by htfujina** [comment]
I would reject this and any other refactoring. The further igbavb driver gets from the Intel igb driver, the harder it gets for me to update it.

**Comment by andrew-elder** [comment]
@htfujina - thanks for the input. Is there a suggested path (location) for this patch to be applied then? How different is igbavb driver from Intel igb driver? Sorry for all the questions. I'm just trying understand the best way forward in terms of policy.

**Comment by htfujina** [comment]
I don't mind bug fixes here, but refactoring for code clarity isn't going to help me much when I do a diff. And I can't accept refactoring on the Intel branch, either, because we don't have the validation resources to test every part that we support.

I think this is the best place for avb fixes.

The igbavb driver is fairly close to the "out-of-tree" igb driver available on sourceforge (and intel.com). Controllers without timestamping support are excised from igbavb and some queues are dedicated to avb. There was also a fix from someone for larger packets.

**Comment by wmarquesr** [comment]
@andrew-elder - I submit this PR for the correct branch as you request.

**Comment by andrew-elder** [comment]
@wmarquesr, it seems the upstream maintainers of this code are not in position to accept re-factoring changes at this time. I apologize for that, but there really isn't anything I can do about it. I'll close this issue in a couple of days unless I get new information.

**Comment by andrew-elder** [comment]
I'm closing this pull request as upstream maintainers are not in a position to accept code refactoring at this time.

**Pull request closed without merging**